tools/toolchain/bolt-fix.patch

```diff
@@ -0,0 +1,47 @@
+From 0f9872ec48b38c4e9ad9cc95144557b15a4c10e5 Mon Sep 17 00:00:00 2001
+Message-Id: <0f9872ec48b38c4e9ad9cc95144557b15a4c10e5.1655215246.git.michal.chojnowski@scylladb.com>
+From: =?UTF-8?q?Micha=C5=82=20Chojnowski?= <michal.chojnowski@scylladb.com>
+Date: Tue, 14 Jun 2022 15:58:56 +0200
+Subject: [PATCH] Fix concurrency bugs in BOLT instrumentation runtime
+
+---
```

Comment:
Did you manage to upstream it?

Reply:
No, I neglected the upstreaming.
I just submitted the patch at https://reviews.llvm.org/D129089.

By the way, yota9 (the maintainer of Aarch64 backend for BOLT) mentioned their "currently internal" implementation of the instrumentation runtime there. I assume it's an ARM implementation. It would be nice if they upstreamed it. We wouldn't have to disable BOLT for ARM then.

Reply:
The BOLT fix has just been merged to upstream.